### ISSUE TITLE
Packer custom image does not use specified service account email.

### ISSUE THREAD

**Issue by keongalvin** [GoogleCloudPlatform/cluster-toolkit#2391]
### Describe the bug

The [service_account_email](https://github.com/GoogleCloudPlatform/hpc-toolkit/blob/main/modules/packer/custom-image/variables.pkr.hcl#L105) option is not propagated to the [toolkit image](https://github.com/GoogleCloudPlatform/hpc-toolkit/blob/main/modules/packer/custom-image/image.pkr.hcl#L94).

### Steps to reproduce

1. Copy any [example](https://github.com/GoogleCloudPlatform/hpc-toolkit/blob/main/community/examples/htc-htcondor.yaml#L48) that uses the packer group, and include a `service_account_email` key in settings.
2. `ghp create <template>` and follow the generated instructions

### Expected behavior

`packer build` uses the specified service_account_email.

### Actual behavior

`packer build` uses the default compute engine service account instead of the supplied service account.

### Version (`ghpc --version`)

```
ghpc version v1.30.0
Built fromn 'main' branch.
Commit info: v1.30.0-0-08ae77e5
```

**Comment by tpdownes** [comment]
@keongalvin this was fixed in #2367 and #2377. If you compile from the `develop` branch, you can obtain this feature now. It will be present in the next official release (which will be v1.30.1 or v1.31.0).

**Comment by tpdownes** [comment]
@keongalvin you should review #2367 carefully as this bug exposed a race condition between the creation of a service account and the actual propagation of IAM permissions that were needed by the Packer build. Our recommendation that you create the service account separately with a period of minutes before using it.

For details:

https://cloud.google.com/iam/docs/access-change-propagation
